libs/sysplugins/smarty_internal_cacheresource_file.php

```diff
@@ -196,8 +196,8 @@ public function clear(Smarty $smarty, $resource_name, $cache_id, $compile_id, $e
      */
     public function hasLock(Smarty $smarty, Smarty_Template_Cached $cached)
     {
-        clearstatcache(true, $cached->lock_id);
-        if (is_file($cached->lock_id)) {
+        clearstatcache(true, $cached->lock_id ?? '');
```

Comment:
you probably don't want to call clearstatcache with empty filename. it's performance killer to clear all cache and looking at the immediate code around this call, doesn't seem you need "clear all cache"

perhaps re-evaluate clearstatcache usage at all, as it's not cached for inexistent files and unlink invalidates automatically:
- https://www.php.net/manual/en/function.clearstatcache.php

Reply:
also, I believe this change should be extracted to own pull request, not related to 8.1 per se, but affects all (currently supported) versions

Reply:
I'm replying here, but I've closed the pr because of cherry-picking the commits from the previous author. Personally I agree with your point, but IMO it's out of scope of this change. What the original author tried to do is "doing whatever php < 8.1 would do". So another pr should cope with this issue.

Reply:
you can still split the commits, fix commit message.

and to preserve authorship (keep credits), add "Co-Authored-By: xxx" git trailer (github renders them ok)

Reply:
Thanks, good to know. I've done something wrong, I guess and now I'm having some commits twice and a merge commit. Looks like I did it wrong using merge instead of using rebase?

Shall we continue here or on the other pr regarding your original comment to clearstatcache?